src/containers/project-overview/index.tsx

```diff
@@ -1,20 +1,59 @@
+import { useEffect, useState } from 'react';
 import { Banner } from '../../components/banner';
 import { Footer } from '../../components/footer';
 import { ProjectsList } from '../../components/project/list';
+import { SupabaseClient } from '@supabase/supabase-js';
+import { Database } from '../../database.types';
+import { useNavigate } from 'react-router-dom';
+
+interface ProjectOverviewContainerProps {
+  supabase: SupabaseClient<Database>;
+}
+
+type Project = Database['public']['Tables']['project']['Row']
+
+export const ProjectOverview: React.FC<ProjectOverviewContainerProps> = ({ supabase }) => {
+
+  const navigate = useNavigate();
+
+  const [projects, setProjects] = useState<Project[]>([]);
+  const [errors, setErrors] = useState<String[]>([]);
+
+  async function handleSignOut() {
+
+    let { error } = await supabase.auth.getSession();
+    if (error)
+      console.warn('Encountered an error whilst fetching user session.', error.cause);
+    else {
+      let { error } = await supabase.auth.signOut();
+      if (error)
+        console.warn('Encountered an error whilst signing out.', error.cause)
+
+      console.log("Okay redirecting to /login...")
+      return navigate("/login");
+    };
+  }
+
+  useEffect(() => {
+
+    async function loadProjects() {
+      const { data, error } = await supabase.from('project').select('*');
+
+      if (data)
+        setProjects(data);
+      else
+        setErrors([error.message]);
+    }
+
+    loadProjects()
+
+  }, []);
 
-export const ProjectOverview = () => {
   return (
     <>
       <Banner />
-      <ProjectsList
-        projects={[
-          { name: 'sierra leone' },
-          { name: 'lebanon' },
-          { name: 'ukraine' },
-          { name: 'turkey' },
-        ]}
-      />
-      <Footer />
+      {errors ? errors.map(error => <p>{error}</p>) : <ProjectsList projects={projects} />}
```

Comment:
This is a bit of a hack to surface errors (if any). Note: errors don't include not being allowed to fetch the data (this is an expected scenario) - they include things like fetching a table that doesn't exist etc.